src/wp-includes/widgets/class-wp-widget-media-image.php

```diff
@@ -239,14 +239,31 @@ public function render_media( $instance ) {
 				$instance['height'] = '';
 			}
 
-			$image = sprintf(
-				'<img class="%1$s" src="%2$s" alt="%3$s" width="%4$s" height="%5$s" />',
-				esc_attr( $classes ),
-				esc_url( $instance['url'] ),
-				esc_attr( $instance['alt'] ),
-				esc_attr( $instance['width'] ),
-				esc_attr( $instance['height'] )
+			$attr = array(
+				'class'    => $classes,
+				'src'      => $instance['url'],
+				'alt'      => $instance['alt'],
+				'width'    => $instance['width'],
+				'height'   => $instance['height'],
+				'decoding' => 'async',
 			);
+
+			$loading_optimization_attr = wp_get_loading_optimization_attributes(
+				'img',
+				$attr,
```

Comment:
```suggestion
				$instance,
```
@spacedmonkey There is a possibility that `$instance` (if modified by a plugin) may carry `loading` and `fetchpriority`, in that case values of `$instance` would take precedence over our logic.

Reply:
$instance contains more than just the attributes for the image. It could contain the title of the widget etc. To be safe, I created a new variable and copied the elements of the array over. 

I think it is safer not to mess with $instance

Reply:
@spacedmonkey Sorry for not being clear in the first instance. 

I was saying if by any chance there is `$instance['loading']` or `$instance['feature']` already present, we must pass it to `wp_get_loading_optimization_attributes`.

You can add it to the newly created `$attr`
 or 
 instead simply pass the`$instance` itself to`wp_get_loading_optimization_attributes()`.
 
 Since you are merging the output `$loading_optimization_attr` with `$attr`, you will not mess with `$instance`.
 
 I feel if someone modifies the widget schema and adds `fetchpriority` and `loading` to the instance using "widget_{id}_instance_schema" filter we must respect those values.
 
@felixarntz  what do you feel?

Reply:
@kt-12 That's a good thought. But since those values cannot be controlled in the widget (there are no UI options for this), it wouldn't be possible to override them, so I think it's not necessary to check for that here.